src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const results = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    let newState;
+
+    switch (action.type) {
+      case 'addProperties':
+        currentState = {
+          ...currentState,
+          ...action.extraData,
+        };
+        newState = { ...currentState };
+        break;
+      case 'removeProperties':
+        newState = { ...currentState };
+
+        for (const key of action.keysToRemove) {
+          delete newState[key];
+        }
+        currentState = { ...newState };
+        break;
+      case 'clear':
+        currentState = {};
+        newState = { ...currentState };
+        break;
+      default:
+        newState = { ...currentState };
+    }
+    results.push(newState);
```

Comment:
you can just make like this
```suggestion
    results.push({...newState});
```